### PR TITLE
refactoring ITDE manager interface

### LINKED ISSUE #93: Refactor ITDE startup functions
See also  
* #89
* #92 

### Acceptance criteria 

In file [exasol/nb_connector/itde_manager.py](https://github.com/exasol/notebook-connector/blob/main/exasol/nb_connector/itde_manager.py) 

* Rename `start_itde()` to `restart_itde()`? 
* Rename method `is_itde_running()` to `is_itde_ready()`. 
* Refactor function  `is_itde_running()` to return an enum rather than a tuple (see below) 

```python
class ItdeContainerStatus(Enum):
    EXISTS = 0
    RUNNING = 1
```
This would also enable to handle the case `not container_name or not network_name` more explicitly by returning `None` or having an explicit enum value `UNKNOWN` or `INVALID`.

### PR DESCRIPTION
**Pull request by ahsimb**
closes #93

### CODE REVIEW

File: test/integration/test_itde_manager.py [hunk 1]
```diff
@@ -145,8 +142,7 @@ def test_take_itde_down_is_not_itde_running(secrets):
     try:
         bring_itde_up(secrets)
         take_itde_down(secrets)
-        itde_exists, itde_running = is_itde_running(secrets)
-        assert not itde_exists
-        assert not itde_running
+        itde_status = get_itde_status(secrets)
+        assert itde_status & ItdeContainerStatus.RUNNING
```
Comment (Nicoretti) [inline test/integration/test_itde_manager.py:146]:
```suggestion
        assert ItdeContainerStatus.RUNNING in itde_status
```

File: test/integration/test_itde_manager.py [hunk 2]
```diff
@@ -105,10 +103,9 @@ def test_itde_start(secrets):
     try:
         bring_itde_up(secrets)
         stop_itde(secrets)
-        start_itde(secrets)
-        itde_exists, itde_running = is_itde_running(secrets)
-        assert itde_exists
-        assert itde_running
+        restart_itde(secrets)
+        itde_status = get_itde_status(secrets)
+        assert itde_status & ItdeContainerStatus.RUNNING
```
Comment (Nicoretti) [inline test/integration/test_itde_manager.py:108]:
```suggestion
        assert ItdeContainerStatus.RUNNING in itde_status
```

File: test/integration/test_itde_manager.py [hunk 3]
```diff
@@ -91,9 +90,8 @@ def test_itde_exists_not_running(secrets):
     try:
         bring_itde_up(secrets)
         stop_itde(secrets)
-        itde_exists, itde_running = is_itde_running(secrets)
-        assert itde_exists
-        assert not itde_running
+        itde_status = get_itde_status(secrets)
+        assert itde_status == ItdeContainerStatus.STOPPED
```
Comment (Nicoretti) [inline test/integration/test_itde_manager.py:94]:
```suggestion
        assert itde_status is ItdeContainerStatus.STOPPED
```

File: test/integration/test_itde_manager.py [hunk 4]
```diff
@@ -70,18 +71,16 @@ def test_itde_exists_and_running(secrets):
...
         remove_itde()
 
 
 def test_itde_neither_exists_nor_running(secrets):
     remove_itde()
-    itde_exists, itde_running = is_itde_running(secrets)
-    assert not itde_exists
-    assert not itde_running
+    itde_status = get_itde_status(secrets)
+    assert itde_status == ItdeContainerStatus.ABSENT
```
Comment (Nicoretti) [inline test/integration/test_itde_manager.py:83]:
```suggestion
    assert itde_status is ItdeContainerStatus.ABSENT
```

File: test/integration/test_itde_manager.py [hunk 5]
```diff
@@ -70,18 +71,16 @@ def test_itde_exists_and_running(secrets):
 
     try:
         bring_itde_up(secrets)
-        itde_exists, itde_running = is_itde_running(secrets)
-        assert itde_exists
-        assert itde_running
+        itde_status = get_itde_status(secrets)
+        assert itde_status & ItdeContainerStatus.RUNNING
```
Comment (Nicoretti) [inline test/integration/test_itde_manager.py:75]:
```suggestion
        assert ItdeContainerStatus.RUNNING in itde_status
```

File: exasol/nb_connector/itde_manager.py [hunk 6]
```diff
@@ -144,61 +154,61 @@ def _get_ipv4_addresses():
...
+        raise RuntimeError("The Docker-DB container doesn't exist.")
+
+    if not (status & ItdeContainerStatus.RUNNING):
+        container_name = conf.get(AILabConfig.itde_container)
+        with ContextDockerClient() as docker_client:
+            container = docker_client.containers.get(container_name)
             container.start()
 
-    _add_current_container_to_db_network(network_name)
+    if not (status & ItdeContainerStatus.VISIBLE):
```
Comment (Nicoretti) [inline exasol/nb_connector/itde_manager.py:209]:
```suggestion
    if ItdeContainerStatus.VISIBLE not in status:
```

File: exasol/nb_connector/itde_manager.py [hunk 7]
```diff
@@ -144,61 +154,61 @@ def _get_ipv4_addresses():
...
+    status = get_itde_status(conf)
 
-    # Start the container using the Docker API, unless it's already running.
-    with ContextDockerClient() as docker_client:
-        container = docker_client.containers.get(container_name)
-        if container.status != 'running':
+    if status == ItdeContainerStatus.ABSENT:
+        raise RuntimeError("The Docker-DB container doesn't exist.")
+
+    if not (status & ItdeContainerStatus.RUNNING):
```
Comment (Nicoretti) [inline exasol/nb_connector/itde_manager.py:203]:
```suggestion
    if ItdeContainerStatus.RUNNING not in status:
```

File: exasol/nb_connector/itde_manager.py [hunk 8]
```diff
@@ -144,61 +154,61 @@ def _get_ipv4_addresses():
...
-    network_name = conf.get(AILabConfig.itde_network)
-    if not container_name or not network_name:
-        raise RuntimeError('Unable to find the name of the Docker container or its network.')
+    status = get_itde_status(conf)
 
-    # Start the container using the Docker API, unless it's already running.
-    with ContextDockerClient() as docker_client:
-        container = docker_client.containers.get(container_name)
-        if container.status != 'running':
+    if status == ItdeContainerStatus.ABSENT:
```
Comment (Nicoretti) [inline exasol/nb_connector/itde_manager.py:200]:
```suggestion
    if status is ItdeContainerStatus.ABSENT:
```

File: exasol/nb_connector/itde_manager.py [hunk 9]
```diff
@@ -144,61 +154,61 @@ def _get_ipv4_addresses():
     return ip_addresses
 
 
-def is_itde_running(conf: Secrets) -> Tuple[bool, bool]:
+def get_itde_status(conf: Secrets) -> ItdeContainerStatus:
```
Comment (Nicoretti) [inline exasol/nb_connector/itde_manager.py:157]:
```suggestion
def itde_status(conf: Secrets) -> ItdeContainerStatus:
```

**Review by Nicoretti — APPROVED** [review]

File: exasol/nb_connector/itde_manager.py (hunk 9 above)
Comment (ahsimb) [reply exasol/nb_connector/itde_manager.py:157]: I think it's better to have a verb in the name.

File: test/integration/test_itde_manager.py (hunk 1 above)
Comment (ahsimb) [reply test/integration/test_itde_manager.py:146]: Also, the status was wrong.

**Review by ckunki — APPROVED** [review]

**Pull request merged**